packages/twig-new/vite.config.ts

```diff
@@ -0,0 +1,135 @@
+import { defineConfig, BuildOptions, Rollup } from "vite";
+import { viteStaticCopy } from "vite-plugin-static-copy";
+import { OutputOptions, rollup, RollupOptions } from "rollup";
+import { nodeResolve } from "@rollup/plugin-node-resolve";
+import commonjs from "@rollup/plugin-commonjs";
+import terser from "@rollup/plugin-terser";
+import swc from "@rollup/plugin-swc";
+import { globSync } from "glob";
+import chalk from "chalk";
+
+import theme from "@ilo-org/themes/tokens/theme/base.json" assert { type: "json" };
+
+const behaviors = globSync("src/components/**/*.behavior.js");
+
+export default defineConfig({
+  plugins: [
+    viteStaticCopy({
+      targets: [
+        {
+          src: "src/components/**/**.component.yml",
+          dest: "components/",
+          rename: (name, extension, fullPath) => {
+            const componentFolder = fullPath.match(
+              /(.*)\/(.*)\/(.*).component.yml$/
+            )?.[2];
+            return `${componentFolder}/${name}.${extension}`;
+          },
+        },
+        {
+          src: "src/components/**/**.twig",
+          dest: "components/",
+          rename: (name, extension, fullPath) => {
+            const componentFolder = fullPath.match(
+              /(.*)\/(.*)\/(.*).twig$/
+            )?.[2];
+            return `${componentFolder}/${name}.${extension}`;
+          },
+          transform: (contents, path) =>
+            contents
+              .toString()
+              .replace(/{{prefix}}/g, theme.themeprefix.value)
+              .replace(/prefix ~/g, `'${theme.themeprefix.value}' ~`),
+        },
+        {
+          src: "node_modules/@ilo-org/styles/css/index.css",
+          dest: "global/",
+          rename: "styles.css",
+        },
+      ],
+    }),
+    {
+      name: "post-build",
+      buildStart() {
+        console.log(
+          `${chalk.green("⚙︎")} ${chalk.yellow("rollup:")} ${chalk.green(
+            "building for production... \n"
+          )}`
+        );
+      },
+      buildEnd() {
+        const grey = chalk.hex("#A5ABB6");
+        console.log(
+          `${chalk.green("✄")} ${chalk.blue("vite:")} ${grey("copied: ")}`
+        );
+        console.log(chalk.green("  - .twig files"));
+        console.log(chalk.green("  - .component.yml files"));
+        console.log(chalk.green("  - global styles\n"));
+
+        for (const behavior of behaviors) {
+          build(behavior);
```

Comment:
It's a bit confusing that the actual build happens in the buildEnd step of Vite.

Reply:
Why? We have the Vite  plugin that hooks into lifecycle of the build step